hack/generator/pkg/astmodel/type_visitor.go

```diff
@@ -155,5 +158,18 @@ func MakeTypeVisitor() TypeVisitor {
 
 			return NewResourceType(visitedSpec, visitedStatus).WithOwner(it.Owner()), nil
 		},
+		VisitArmType: func(this *TypeVisitor, at *ArmType, ctx interface{}) (Type, error) {
+			newType, err := this.Visit(&at.objectType, ctx)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to visit ARM underlying type %v", at.objectType)
+			}
+
+			ot, ok := newType.(*ObjectType)
+			if !ok {
+				return nil, errors.Wrapf(err, "expected transformation of ARM underlying type %s to return ObjectType, not %v", at.objectType, ot)
```

Comment:
This should be `errors.Errorf` rather than `errors.Wrapf` - `Wrapf` returns nil if the error  being wrapped is nil, which it always will be here.

Reply:
Good catch, thanks.